Xamarin.Essentials/AppInfo/AppInfo.ios.tvos.watchos.mac.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Foundation;
+#if __IOS__ || __TVOS__
+using UIKit;
+#endif
+
+namespace Xamarin.Essentials
+{
+    public static partial class AppInfo
+    {
+        static string PlatformGetPackageName() => GetBundleValue("CFBundleIdentifier");
+
+        static string PlatformGetName() => GetBundleValue("CFBundleDisplayName") ?? GetBundleValue("CFBundleName");
+
+        static string PlatformGetVersionString() => GetBundleValue("CFBundleShortVersionString");
+
+        static string PlatformGetBuild() => GetBundleValue("CFBundleVersion");
+
+        static string GetBundleValue(string key)
+           => NSBundle.MainBundle.ObjectForInfoDictionary(key)?.ToString();
+
+#if __IOS__ || __TVOS__
+        static void PlatformShowSettingsUI() =>
+            UIApplication.SharedApplication.OpenUrl(new NSUrl(UIApplication.OpenSettingsUrlString));
+#else
+        static void PlatformShowSettingsUI() =>
+            throw new FeatureNotSupportedException();
+#endif
+
+#if __IOS__ || __TVOS__
+        static AppTheme PlatformRequestedTheme()
+        {
+            if (!Platform.HasOSVersion(13, 0))
+                return AppTheme.Unspecified;
+
+            var uiStyle = Platform.GetCurrentUIViewController()?.TraitCollection?.UserInterfaceStyle ??
+                UITraitCollection.CurrentTraitCollection.UserInterfaceStyle;
+
+            return uiStyle switch
+            {
+                UIUserInterfaceStyle.Light => AppTheme.Light,
+                UIUserInterfaceStyle.Dark => AppTheme.Dark,
+                _ => AppTheme.Unspecified
+            };
+        }
+#else
+        static AppTheme PlatformRequestedTheme() =>
```

Comment:
I think Cocoa can do this as well? https://github.com/xamarin/xamarin-macios/blob/master/src/appkit.cs#L380

Reply:
Awesome. Added.